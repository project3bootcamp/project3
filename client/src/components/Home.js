import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import Button from '@material-ui/core/Button';
import Typography from '@material-ui/core/Typography';
import Grid from '@material-ui/core/Grid';
import Paper from '@material-ui/core/Paper';
import black from '@material-ui/core/colors/grey';
import AutoSave from './AutoSave';

export default class Home extends Component {
    state = {
        searches: [],
    };
    componentDidMount(){
        this.getSearches();
        
    }
    getSearches = () => {
        const savedSearches = new AutoSave();
        savedSearches.onGetSearched().then(res => {
        console.log("getting user saved searches");
        console.log(`Searches for ${res.data.name} are being loaded`);
        this.setState({ searches:res.data.search })
        console.log(res.data.search);
        });
    }
    handleClick = () => {
        console.log('this is:', this);
      }
    
      render() {
        return (
            <div>
                User Profile
            <Grid container spacing={32} >
              <Grid item xs={12}>
                <Grid container justify="center" spacing={8} >
                    {!this.state.searches.length ?(<Typography variant='h4'>No Searches Found!</Typography>) :
                
                       ( this.state.searches.map((search, index) => {
                        return (
                          <Grid item xs={12}container
                          direction="column"
                          justify="center"
                          alignItems="center">
                            <Paper>
                                <Button >{search.actor1},{search.actor2}</Button>
                            </Paper>
                          </Grid>
                        )}
                       )
                    )}
                    <Grid item xs={6}>
                      <Paper elevation={1}>
                        <Link style={{color: black[900]}} to="/actorsMatch"> Actors Match</Link>
                      </Paper>
                    </Grid>
                </Grid>
              </Grid>
            </Grid>
            </div>
        );
      }
    }