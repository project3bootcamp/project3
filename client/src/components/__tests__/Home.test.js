import { shallow } from 'enzyme';
import React from 'react';
import Home from '../Home';

describe('Home', () => {
    it('renders correctly', () => {
      const wrapper = shallow(<Home />);
      expect(wrapper).toMatchSnapshot();
      
    });
  });