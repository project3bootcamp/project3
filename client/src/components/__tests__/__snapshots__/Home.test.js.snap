// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Home />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        "User Profile",
        <WithStyles(Grid)
          container={true}
          spacing={32}
        >
          <WithStyles(Grid)
            item={true}
            xs={12}
          >
            <WithStyles(Grid)
              container={true}
              justify="center"
              spacing={8}
            >
              <WithStyles(Typography)
                variant="h4"
              >
                No Searches Found!
              </WithStyles(Typography)>
              <WithStyles(Grid)
                item={true}
                xs={6}
              >
                <WithStyles(Paper)
                  elevation={1}
                >
                  <Link
                    replace={false}
                    style={
                      Object {
                        "color": "#212121",
                      }
                    }
                    to="/actorsMatch"
                  >
                     Actors Match
                  </Link>
                </WithStyles(Paper)>
              </WithStyles(Grid)>
            </WithStyles(Grid)>
          </WithStyles(Grid)>
        </WithStyles(Grid)>,
      ],
    },
    "ref": null,
    "rendered": Array [
      "User Profile",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <WithStyles(Grid)
            item={true}
            xs={12}
          >
            <WithStyles(Grid)
              container={true}
              justify="center"
              spacing={8}
            >
              <WithStyles(Typography)
                variant="h4"
              >
                No Searches Found!
              </WithStyles(Typography)>
              <WithStyles(Grid)
                item={true}
                xs={6}
              >
                <WithStyles(Paper)
                  elevation={1}
                >
                  <Link
                    replace={false}
                    style={
                      Object {
                        "color": "#212121",
                      }
                    }
                    to="/actorsMatch"
                  >
                     Actors Match
                  </Link>
                </WithStyles(Paper)>
              </WithStyles(Grid)>
            </WithStyles(Grid)>
          </WithStyles(Grid)>,
          "container": true,
          "spacing": 32,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <WithStyles(Grid)
              container={true}
              justify="center"
              spacing={8}
            >
              <WithStyles(Typography)
                variant="h4"
              >
                No Searches Found!
              </WithStyles(Typography)>
              <WithStyles(Grid)
                item={true}
                xs={6}
              >
                <WithStyles(Paper)
                  elevation={1}
                >
                  <Link
                    replace={false}
                    style={
                      Object {
                        "color": "#212121",
                      }
                    }
                    to="/actorsMatch"
                  >
                     Actors Match
                  </Link>
                </WithStyles(Paper)>
              </WithStyles(Grid)>
            </WithStyles(Grid)>,
            "item": true,
            "xs": 12,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <WithStyles(Typography)
                  variant="h4"
                >
                  No Searches Found!
                </WithStyles(Typography)>,
                <WithStyles(Grid)
                  item={true}
                  xs={6}
                >
                  <WithStyles(Paper)
                    elevation={1}
                  >
                    <Link
                      replace={false}
                      style={
                        Object {
                          "color": "#212121",
                        }
                      }
                      to="/actorsMatch"
                    >
                       Actors Match
                    </Link>
                  </WithStyles(Paper)>
                </WithStyles(Grid)>,
              ],
              "container": true,
              "justify": "center",
              "spacing": 8,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "No Searches Found!",
                  "variant": "h4",
                },
                "ref": null,
                "rendered": "No Searches Found!",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <WithStyles(Paper)
                    elevation={1}
                  >
                    <Link
                      replace={false}
                      style={
                        Object {
                          "color": "#212121",
                        }
                      }
                      to="/actorsMatch"
                    >
                       Actors Match
                    </Link>
                  </WithStyles(Paper)>,
                  "item": true,
                  "xs": 6,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": <Link
                      replace={false}
                      style={
                        Object {
                          "color": "#212121",
                        }
                      }
                      to="/actorsMatch"
                    >
                       Actors Match
                    </Link>,
                    "elevation": 1,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": " Actors Match",
                      "replace": false,
                      "style": Object {
                        "color": "#212121",
                      },
                      "to": "/actorsMatch",
                    },
                    "ref": null,
                    "rendered": " Actors Match",
                    "type": [Function],
                  },
                  "type": [Function],
                },
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "User Profile",
          <WithStyles(Grid)
            container={true}
            spacing={32}
          >
            <WithStyles(Grid)
              item={true}
              xs={12}
            >
              <WithStyles(Grid)
                container={true}
                justify="center"
                spacing={8}
              >
                <WithStyles(Typography)
                  variant="h4"
                >
                  No Searches Found!
                </WithStyles(Typography)>
                <WithStyles(Grid)
                  item={true}
                  xs={6}
                >
                  <WithStyles(Paper)
                    elevation={1}
                  >
                    <Link
                      replace={false}
                      style={
                        Object {
                          "color": "#212121",
                        }
                      }
                      to="/actorsMatch"
                    >
                       Actors Match
                    </Link>
                  </WithStyles(Paper)>
                </WithStyles(Grid)>
              </WithStyles(Grid)>
            </WithStyles(Grid)>
          </WithStyles(Grid)>,
        ],
      },
      "ref": null,
      "rendered": Array [
        "User Profile",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <WithStyles(Grid)
              item={true}
              xs={12}
            >
              <WithStyles(Grid)
                container={true}
                justify="center"
                spacing={8}
              >
                <WithStyles(Typography)
                  variant="h4"
                >
                  No Searches Found!
                </WithStyles(Typography)>
                <WithStyles(Grid)
                  item={true}
                  xs={6}
                >
                  <WithStyles(Paper)
                    elevation={1}
                  >
                    <Link
                      replace={false}
                      style={
                        Object {
                          "color": "#212121",
                        }
                      }
                      to="/actorsMatch"
                    >
                       Actors Match
                    </Link>
                  </WithStyles(Paper)>
                </WithStyles(Grid)>
              </WithStyles(Grid)>
            </WithStyles(Grid)>,
            "container": true,
            "spacing": 32,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <WithStyles(Grid)
                container={true}
                justify="center"
                spacing={8}
              >
                <WithStyles(Typography)
                  variant="h4"
                >
                  No Searches Found!
                </WithStyles(Typography)>
                <WithStyles(Grid)
                  item={true}
                  xs={6}
                >
                  <WithStyles(Paper)
                    elevation={1}
                  >
                    <Link
                      replace={false}
                      style={
                        Object {
                          "color": "#212121",
                        }
                      }
                      to="/actorsMatch"
                    >
                       Actors Match
                    </Link>
                  </WithStyles(Paper)>
                </WithStyles(Grid)>
              </WithStyles(Grid)>,
              "item": true,
              "xs": 12,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <WithStyles(Typography)
                    variant="h4"
                  >
                    No Searches Found!
                  </WithStyles(Typography)>,
                  <WithStyles(Grid)
                    item={true}
                    xs={6}
                  >
                    <WithStyles(Paper)
                      elevation={1}
                    >
                      <Link
                        replace={false}
                        style={
                          Object {
                            "color": "#212121",
                          }
                        }
                        to="/actorsMatch"
                      >
                         Actors Match
                      </Link>
                    </WithStyles(Paper)>
                  </WithStyles(Grid)>,
                ],
                "container": true,
                "justify": "center",
                "spacing": 8,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "No Searches Found!",
                    "variant": "h4",
                  },
                  "ref": null,
                  "rendered": "No Searches Found!",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": <WithStyles(Paper)
                      elevation={1}
                    >
                      <Link
                        replace={false}
                        style={
                          Object {
                            "color": "#212121",
                          }
                        }
                        to="/actorsMatch"
                      >
                         Actors Match
                      </Link>
                    </WithStyles(Paper)>,
                    "item": true,
                    "xs": 6,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": <Link
                        replace={false}
                        style={
                          Object {
                            "color": "#212121",
                          }
                        }
                        to="/actorsMatch"
                      >
                         Actors Match
                      </Link>,
                      "elevation": 1,
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "children": " Actors Match",
                        "replace": false,
                        "style": Object {
                          "color": "#212121",
                        },
                        "to": "/actorsMatch",
                      },
                      "ref": null,
                      "rendered": " Actors Match",
                      "type": [Function],
                    },
                    "type": [Function],
                  },
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
